pkg/machine/ignition/ignition.go

```diff
@@ -743,6 +776,37 @@ func (i *IgnitionBuilder) Build() error {
 	return i.dynamicIgnition.Write()
 }
 
+func GetNetRecoveryFile() string {
+	return `#!/bin/bash
+# Verify network health, and bounce the network device if host connectivity
+# is lost. This is a temporary workaround for a known rare qemu/virtio issue
+# that affects some systems
+
+sleep 120 # allow time for network setup on initial boot
+while true; do
+  sleep 30
+  curl -s -o /dev/null --connect-timeout 10 http://192.168.127.1/health
+  if [ "$?" != "0" ]; then
+    echo "bouncing nic due to loss of connectivity with host"
+    ifconfig enp0s1 down; ifconfig enp0s1 up
+  fi
+done
+`
+}
+
+func GetNetRecoveryUnitFile() *parser.UnitFile {
+	recoveryUnit := parser.NewUnitFile()
+	recoveryUnit.Add("Unit", "Description", "Verifies health of network and recovers if necessary")
+	recoveryUnit.Add("Unit", "After", "sshd.socket sshd.service")
```

Comment:
Fwiw, I'm not sure `sshd.service` is required here?

Reply:
sshd.socket and sshd.service are mutually exclusive alternates (currently our fcos images are using sshd.service atm. Our other units are declared as after both (my assumption is to be compatible if the base images witch to the inet socket approach in the future), so just mirroring that pattern here.

Reply:
Ah ok, I assumed the FCOS images used `sshd.socket` already.